### ISSUE TITLE
New learning function

### ISSUE THREAD

**Issue by lenhoanglnh** [tournesol-app/tournesol#88]
Instead of the Bradley-Terry cost, implement this new loss.
From my theoretical analysis, this will make the scores a lot more interpretable and more stable.
(here sinh is the hyperbolic sine function)

<img width="1044" alt="Screen Shot 2021-05-05 at 12 18 34" src="https://user-images.githubusercontent.com/10453308/117127153-07112680-ad9c-11eb-9a2f-b3c6b5988fb9.png">

Intuitively, it means that when a contributor provides a rating 0.5, it is new as though they rated the video 4 times, and said that the right video was better 3 times out of 4, and that they preferred the left video one time out of 4 (but the combined rating is "worth" one rating).

More interestingly, this yields the following translation between ratings and score differences (typically, a rating = 0.5 that says B is better than A corresponds to a score difference of ≈ 1.9).
<img width="1791" alt="Screen Shot 2021-05-14 at 16 08 13" src="https://user-images.githubusercontent.com/10453308/118282466-97ebae80-b4ce-11eb-9e44-b14645a9a002.png">
https://www.desmos.com/calculator/km7v1aajiw

**Comment by sergeivolodin** [comment]
Can you write both the old equation (full loss) and the new one one below another?

**Comment by lenhoanglnh** [comment]
So here's the old loss (I forgot to mention, here, \theta_{ab} = \theta_a - \theta_b). 
Also, the scores here also all depend on the contributor.
<img width="221" alt="Screen Shot 2021-05-06 at 14 45 19" src="https://user-images.githubusercontent.com/10453308/117300286-b243dd80-ae79-11eb-8a30-6403da3424bf.png">

Here's the global loss function:
<img width="453" alt="Screen Shot 2021-05-06 at 14 49 23" src="https://user-images.githubusercontent.com/10453308/117300832-40b85f00-ae7a-11eb-956d-d9e18e377d38.png">
The modification occurs in the first term.

In the code, here's the line to be changed:
https://github.com/tournesol-app/tournesol/blob/d1ea087b3ac42ac6aca18a428e150ad1ed006fde/backend/backend/ml_model/preference_aggregation_featureless_tf_sparse.py#L145
I'm not sure if tensorflow has a built-in function "hyperbolic sine function". It would be better as there is otherwise a (false) issue at t=0...

**Comment by lfaucon** [comment]
I don't think this issue should have prio:critical 
- It does not block usage of the platform
- It is not a security or bad recommendation issue

**Comment by lenhoanglnh** [comment]
It is needed for the paper though 😅

**Comment by sergeivolodin** [comment]
Decided to get the results and not use it, downgrading to minor
